core/src/main/java/org/apache/accumulo/core/client/impl/ClientConfConverter.java

```diff
@@ -229,6 +239,37 @@ public void getProperties(Map<String,String> props, Predicate<String> filter) {
         return null;
       }
     };
+  }
+
+  public static ClientConfiguration toClientConf(AccumuloConfiguration conf) {
+    ClientConfiguration clientConf = ClientConfiguration.create();
 
+    // Servers will only have the full principal in their configuration -- parse the
+    // primary and realm from it.
+    final String serverPrincipal = conf.get(Property.GENERAL_KERBEROS_PRINCIPAL);
+
+    final KerberosName krbName;
+    try {
+      krbName = new KerberosName(serverPrincipal);
+      clientConf.setProperty(ClientConfiguration.ClientProperty.KERBEROS_SERVER_PRIMARY, krbName.getServiceName());
+    } catch (Exception e) {
```

Comment:
This seems overly broad, is there something more specific to do? Could empty case be checked for?  Is there a more specific exception to catch?  Also could log exception at debug.

Reply:
I added check for empty case in 64f260f0b10013c4cd and removed catch. What do you think?

Reply:
I don't have a deep understanding of the code, but it looks less likely to hide problems.